t/29-backend-svirt.t

```diff
@@ -0,0 +1,120 @@
+use Test::Fatal;
+use Test::MockModule;
+use Test::MockObject;
+use Test::Mock::Time;
+use Test::Output qw(stderr_like);
+use Mojo::File qw(tempdir);
+use Mojo::Util qw(scope_guard);
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use backend::svirt;
+use distribution;
+
+my $dir = tempdir("/tmp/$FindBin::Script-XXXX");
+chdir $dir;
+my $cleanup = scope_guard sub { chdir $Bin; undef $dir };
+
+my $ssh_object = Test::MockObject->new();
+$ssh_object->set_true(qw/disconnect scp_get blocking/);
+$ssh_object->set_always('error', 'Mock SSH Error');
+
+my $chan_object = Test::MockObject->new();
+$chan_object->set_true(qw/exec/);
+
+my $run_ssh_cmd_mock = Test::MockModule->new('backend::baseclass');
+$run_ssh_cmd_mock->redefine(new_ssh_connection => sub ($self, %args) { return $ssh_object });
+$run_ssh_cmd_mock->redefine(start_ssh_serial => sub { return ($ssh_object, $chan_object) });
+
+sub redefine_ssh ($excepted = 0) {
+    $run_ssh_cmd_mock->redefine(run_ssh_cmd => sub ($self, $cmd, %args) {
+            return ref($excepted) eq 'ARRAY' ? @$excepted : $excepted;
+    });
+}
+
+$bmwqemu::vars{WORKER_HOSTNAME} = 'localhost';
+$bmwqemu::vars{VIRSH_HOSTNAME} = 'foobar';
+$bmwqemu::vars{VIRSH_USERNAME} = 'root';
+$bmwqemu::vars{VIRSH_PASSWORD} = 'password';
+$bmwqemu::vars{JOBTOKEN} = "JobToken";
+
+my $distri = $testapi::distri = distribution->new();
+
+subtest 'Generic svirt backend' => sub {
+    my $backend = backend::svirt->new;
+    my $bmwqemu_mock = Test::MockModule->new('bmwqemu');
+    # silence some log output for cleaner tests
+    $bmwqemu_mock->noop('diag');
+    $bmwqemu_mock->noop('log_call');
+    redefine_ssh;
+    $backend->{need_delete_log} = 1;
+    ok $backend->do_start_vm, 'can start vm';
+    # not kvm/hyperv or vmware
+    is $backend->can_handle({function => 'snapshots'}), undef, 'can not handle snapshots';
+    is $backend->save_snapshot({name => "Snap1"}), undef, 'can save snapshot';
+    is $backend->load_snapshot({name => "Snap1"}), '', 'can load snapshot';
+    ok $backend->start_serial_grab('svirt'), 'can start serial grab';
+    ok $backend->do_stop_vm, 'can stop vm';
+    redefine_ssh "Power OFF";
+    is $backend->is_shutdown, 'Power OFF', 'can call is_shutdown';
+};
+
+subtest 'VMWARE backend' => sub {
+    $bmwqemu::vars{VIRSH_VMM_FAMILY} = 'vmware';
+    $bmwqemu::vars{VMWARE_HOST} = "foobar";
+    $bmwqemu::vars{VMWARE_SERIAL_PORT} = "222";
+
+    my $backend = backend::svirt->new;
+    my $bmwqemu_mock = Test::MockModule->new('bmwqemu');
+    # silence some log output for cleaner tests
+    $bmwqemu_mock->noop('diag');
+    $bmwqemu_mock->noop('log_call');
+    redefine_ssh;
+    $backend->{need_delete_log} = 1;
+    ok $backend->do_start_vm, 'can start vm';
+    is $backend->can_handle({function => 'snapshots'})->{ret}, 1, 'can handle snapshots';
+    is $backend->save_snapshot({name => "Snap1"}), undef, 'can save snapshot';
```

Comment:
```suggestion
    is $backend->save_snapshot({name => "Snap1"}), 1, 'can save snapshot';
```
If it doesn't return a truthy value then either change the function accordingly or remove the check (which is otherwise quite pointless).

**The same counts for further occurrences.**

Reply:
> If it doesn't return a truthy value then either change the function accordingly or remove the check (which is otherwise quite pointless).

change backend functions ? so change behavior to be different for rest of backends ?

Reply:
Right, if that would make it inconsistent with the other backends than better don't change it. However, maybe it still makes sense to remove the check and just invoke the function (if the return value isn't really telling us mach here). Maybe you can add a better check to see whether `save_snapshot` had actually the expected effect, e.g. whether `run_ssh_cmd` has been invoked as expected.